src/lib/optimizer/strategy/chunk_pruning_rule.cpp

```diff
@@ -282,7 +284,8 @@ std::set<ChunkID> ChunkPruningRule::_compute_exclude_list(
     }
 
     // Cache result
-    _excluded_chunk_ids_by_predicate_node_cache.emplace(predicate_node, current_excluded_chunk_ids);
+    _excluded_chunk_ids_by_predicate_node_cache.emplace(std::make_pair(stored_table_node, predicate_node),
```

Comment:
Are you sure you need make_pair here? I feel like `emplace` *might* be able to do this automatically.

Reply:
Good point, thanks!

Reply:
Unfortunately, I do not get this to work. https://stackoverflow.com/questions/25766730/emplacing-a-stdpair

Reply:
Never mind then - just a quick thought.